sklearn/model_selection/_search.py

```diff
@@ -918,15 +1005,32 @@ class RandomizedSearchCV(BaseSearchCV):
 
     Attributes
     ----------
-    grid_scores_ : list of named tuples
-        Contains scores for all parameter combinations in param_grid.
-        Each entry corresponds to one parameter setting.
-        Each named tuple has the attributes:
-
-            * ``parameters``, a dict of parameter settings
-            * ``mean_validation_score``, the mean score over the
-              cross-validation folds
-            * ``cv_validation_scores``, the list of scores for each fold
+    search_results_ : dict of numpy (masked) ndarrays
+        A dict with keys as column headers and values as columns, that can be
+        imported into a pandas DataFrame.
+
+        For instance the below given table
+
+        kernel|gamma|degree|accuracy_score_split_0...|accuracy_score_mean ...|
+        =====================================================================
+        'poly'|  -  |  2   |           0.8           |         0.81          |
+        'poly'|  -  |  3   |           0.7           |         0.60          |
+        'rbf' | 0.1 |  -   |           0.8           |         0.75          |
+        'rbf' | 0.2 |  -   |           0.9           |         0.82          |
+
+        will be represented by a search_results_ dict of :
+
+        {'kernel' : masked_array(data = ['poly', 'poly', 'rbf', 'rbf'],
+                                 mask = [False False False False]...)
+         'gamma' : masked_array(data = [-- -- 0.1 0.2],
+                                mask = [ True  True False False]...),
+         'degree' : masked_array(data = [2.0 3.0 -- --],
+                                 mask = [False False  True  True]...),
+         'accuracy_score_split_0' : [0.8, 0.7, 0.8, 0.9],
```

Comment:
I guess that's not terrible...

On 1 May 2016 at 16:29, Henry Lin <notifications@github.com> wrote:

> In sklearn/model_selection/_search.py
> <https://github.com/scikit-learn/scikit-learn/pull/6697#discussion_r61681497>
> :
>
> > +        kernel|gamma|degree|accuracy_score_split_0...|accuracy_score_mean ...|
> > +        =====================================================================
> > +        'poly'|  -  |  2   |           0.8           |         0.81          |
> > +        'poly'|  -  |  3   |           0.7           |         0.60          |
> > +        'rbf' | 0.1 |  -   |           0.8           |         0.75          |
> > +        'rbf' | 0.2 |  -   |           0.9           |         0.82          |
> > +
> > +        will be represented by a search_results_ dict of :
> > +
> > +        {'kernel' : masked_array(data = ['poly', 'poly', 'rbf', 'rbf'],
> > +                                 mask = [False False False False]...)
> > +         'gamma' : masked_array(data = [-- -- 0.1 0.2],
> > +                                mask = [ True  True False False]...),
> > +         'degree' : masked_array(data = [2.0 3.0 -- --],
> > +                                 mask = [False False  True  True]...),
> > +         'accuracy_score_split_0' : [0.8, 0.7, 0.8, 0.9],
>
> @jnothman <https://github.com/jnothman> Assuming that the folds in
> question are named accuracy_score_split_0, ..., accuracy_score_split_(k-1),
> and the pandas dataframe with the scores is called search_results_, it's
> as easy as
>
> fold_scores = search_results_.select(lambda name: name.startswith("accuracy_score_split_"), axis=1)
> search_results_["std"] = fold_scores.std(axis=1)  # To place it into the dataframe
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/scikit-learn/scikit-learn/pull/6697/files/e153cbbbe7fc8ae5edbb427ca6f3e22d63bd6f0a#r61681497>
>